### PR TITLE
Throttle bulk code issuance using server ratelimiter config

### PR DESCRIPTION
**Pull request by whaught**
Issue https://github.com/google/exposure-notifications-verification-server/issues/1061

## Proposed Changes

* Look for the retry-after header on error response
* if found, sleep as directed

**Release Note**

```release-note
Throttle batches for bulk code issuance
```

### CODE REVIEW

**Comment by sethvargo** [comment]
> Alternative: instead of spreading load evenly we could burst up to remaining tokens and then sleep until the interval. Would make things faster for small CSVs that are under the count/interval. That's a little trickier because we'd actually need to peek into the ratelimiter (not just the config)

I would prefer this approach, but you don't need to peek into the limiter, look at the response headers. They'll include all the data you need 👍

**Comment by whaught** [comment]
> > Alternative: instead of spreading load evenly we could burst up to remaining tokens and then sleep until the interval. Would make things faster for small CSVs that are under the count/interval. That's a little trickier because we'd actually need to peek into the ratelimiter (not just the config)
> 
> I would prefer this approach, but you don't need to peek into the limiter, look at the response headers. They'll include all the data you need 👍

I just saw that we have `X-Retry-After` in the api doc, which is what I need for this

**Comment by mikehelmick** [comment]
> > > Alternative: instead of spreading load evenly we could burst up to remaining tokens and then sleep until the interval. Would make things faster for small CSVs that are under the count/interval. That's a little trickier because we'd actually need to peek into the ratelimiter (not just the config)
> > 
> > 
> > I would prefer this approach, but you don't need to peek into the limiter, look at the response headers. They'll include all the data you need 👍
> 
> I just saw that we have `X-Retry-After` in the api doc, which is what I need for this

one consideration with the consuming all of the realm's quota - it makes it so that online code distribution won't work during this. probably not a big deal

File: docs/api.md [hunk 1]
```diff
@@ -336,7 +336,7 @@ You can expect the following responses from this API:
 
 -   `412` - The client requested a precondition that cannot be satisfied.
 
--   `429` - The client is rate limited. Check the `X-Retry-After` header to
```
Comment (whaught) [inline docs/api.md:339]: There is no `x-` for retry-after
https://github.com/sethvargo/go-limiter/blob/86d882305d36ae8c59138bc9a93b3599f0b70936/httplimit/middleware.go#L28

Comment (sethvargo) [reply docs/api.md:339]: Why would I remember my own code? 😝

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**